### ISSUE TITLE
Migrator and consistency checker should be separate

### ISSUE THREAD

**Issue by bserdar** [lightblue-platform/lightblue-migrator#186]
We need two deamons:
  - Consistency checker. This should look at migrated data, retrieve original data, compare, and report inconsistencies. Optionally it should remigrate them.
 - Migrator: This should read what needs to be migrated from the db, and migrate them.

**Comment by dcrissman** [comment]
What is the advantage of two over one?

Would the "migrator" assume the entity on the target lightblue instance is a blank slate?

**Comment by bserdar** [comment]
In my opinion, they are two different things: migrator looks at the
original, and migrates any necessary changes. Consistency checker looks at
the migrated data, and compares it to the original.

**Comment by dcrissman** [comment]
So your use case might be, you want to be able to check the consistency ad-hoc but without actually changing anything, sort of like you might run unit tests for assurances that everything still works?

> Consistency checker looks at the migrated data, and compares it to the original.

Migrating data involves some conversion. It might be difficult reversing that conversion, especially if the source was not a lightblue instance.

**Comment by bserdar** [comment]
On Thu, May 14, 2015 at 9:42 AM, Dennis Crissman <notifications@github.com>
wrote:

> So your use case might be, you want to be able to check the consistency
> ad-hoc but without actually changing anything, sort of like you might run
> unit tests for assurances that everything still works?
>

I don't think I got this right, but we decided that consistency checker
fixes things if it finds inconsistencies. So, if there is an inconsistency,
consistency checker would create a new migrator job for inconsistent
entities, and then migrator would re-migrate them.


>  Consistency checker looks at the migrated data, and compares it to the
> original.
>
> Migrating data involves some conversion. It might be difficult reversing
> that conversion, especially if the source was not a lightblue instance.
>

There is no need for reversion, the only capability required here is to be
able to retrieve original data. Consistency checker would retrieve
lightblue data, then it retrieves the matching original, converts the
original to lightblue, and compares the two. If they are the same, then
there is no problem.
